### PR TITLE
Simplify Connection Configuration

### PR DESCRIPTION
**Pull request by q-src**
This PR simplifies the RabbitMQ and JMS strategies by the following changes:
* Builder pattern introduced, which allows building connection by calling builder methods or by just passing a `Properties` instance.
* RabbitMQ: Required specification of a queue name removed. RabbitMQ will create a random temporary queue per default.

### CODE REVIEW

File: src/main/java/com/github/libgraviton/messaging/config/ContextProperties.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    /**
+     * Constructor.
+     *
+     * @param properties The properties instance holding all properties.
+     * @param context The config to be used.
+     */
+    public ContextProperties(Properties properties, String context) {
+        super(properties);
+        this.context = context + '.';
```
Comment (wegenmic) [inline src/main/java/com/github/libgraviton/messaging/config/ContextProperties.java]: Should we really manipulate a context we get as an input param?

File: src/main/java/com/github/libgraviton/messaging/config/PropertyUtil.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package com.github.libgraviton.messaging.config;
+
+import java.util.Properties;
+
+/**
+ * Created by tgdpaad2 on 15/02/17.
+ */
+public class PropertyUtil {
```
Comment (wegenmic) [inline src/main/java/com/github/libgraviton/messaging/config/PropertyUtil.java]: Add missing javadoc for PropertyUtil class

File: src/main/java/com/github/libgraviton/messaging/strategy/jms/JmsConnection.java [hunk 3]
```diff
@@ -159,4 +148,56 @@ protected void closeConnection() throws CannotCloseConnection {
...
+        /**
+         * Defines the JMS connection factory.
+         *
+         * @param connectionFactory The connection factory
+         *
+         * @return self
+         */
+        public JmsBuilder connectionFactory(ConnectionFactory connectionFactory) {
+            this.connectionFactory = connectionFactory;
+            return (JmsBuilder) this;
```
Comment (wegenmic) [inline src/main/java/com/github/libgraviton/messaging/strategy/jms/JmsConnection.java]: we should try to avoid unchecked casts when returning 'this'.

**Comment by wegenmic** [comment]
In general we should use the modifiers 'static final' for constants (normally written in capital letters)

File: src/main/java/com/github/libgraviton/messaging/strategy/jms/JmsConnection.java (hunk 3 above)
Comment (wegenmic) [inline src/main/java/com/github/libgraviton/messaging/strategy/jms/JmsConnection.java]: same in class QueueConnection

File: src/main/java/com/github/libgraviton/messaging/config/ContextProperties.java (hunk 1 above)
Comment (q-src) [inline src/main/java/com/github/libgraviton/messaging/config/ContextProperties.java]: The idea is that you don't need to suffix your context by `.`, which would most likely always be the case as `.` is the default delimiter for property names (at least in our libs). How about a `setContextDelimiter()`?

File: src/main/java/com/github/libgraviton/messaging/strategy/jms/JmsConnection.java (hunk 3 above)
Comment (q-src) [inline src/main/java/com/github/libgraviton/messaging/strategy/jms/JmsConnection.java]: The only solution I found is this one: http://stackoverflow.com/questions/17164375/subclassing-a-java-builder-class/34741836#34741836. I wasn't really happy with this either. 
What do you vote for, `getThis()` or unchecked casts?

**Pull request merged**